tests/functional/sources/test_source_fresher_state.py

```diff
@@ -0,0 +1,727 @@
+import os
+import json
+import shutil
+import pytest
+from datetime import datetime, timedelta
+
+from dbt.exceptions import InternalException
+
+
+import dbt.version
+from tests.functional.sources.common_source_setup import BaseSourcesTest
+
+from tests.functional.sources.fixtures import (
+    error_models__schema_yml,
+    models__newly_added_model_sql,
+    models__newly_added_error_model_sql,
+)
+
+
+# TODO: We may create utility classes to handle reusable fixtures.
+def copy_to_previous_state():
+    shutil.copyfile("target/manifest.json", "previous_state/manifest.json")
+    shutil.copyfile("target/run_results.json", "previous_state/run_results.json")
+
+
+# put these here for now to get tests working
+class AnyStringWith:
+    def __init__(self, contains=None):
+        self.contains = contains
+
+    def __eq__(self, other):
+        if not isinstance(other, str):
+            return False
+
+        if self.contains is None:
+            return True
+
+        return self.contains in other
+
+    def __repr__(self):
+        return "AnyStringWith<{!r}>".format(self.contains)
+
+
+class AnyFloat:
+    """Any float. Use this in assertEqual() calls to assert that it is a float."""
+
+    def __eq__(self, other):
+        return isinstance(other, float)
```

Comment:
These were copied from https://github.com/dbt-labs/dbt-core/pull/4935. Maybe they eventually want to move into `dbt/core/tests/util.py`? (cc @gshank)